src/scheduler/server_info.cpp

```diff
@@ -996,6 +996,34 @@ free_server_psets(std::vector<server_psets>& spsets)
 	}
```

Comment:
Don't we also need to call spsets.clear() which actually frees the other element svr_inst_id as well and makes the size of vector to zero ?

Reply:
No, since we use "new" and "delete" to allocate/deallocate server_info now, it'll automatically clear any vectors itself.

Reply:
When I just checked in gdb right after free_server_psets(sinfo->svr_to_psets) gets called, I am getting the following. This is the reason I suspected that it is not completely freed. Just double check once. Maybe gdb is showing wrong. 
(gdb) p	sinfo->svr_to_psets
$3 = std::vector of length 2, capacity 2 = {{svr_inst_id = "spark:15001", np = 0x14cb400}, {svr_inst_id = "spark2:15001", 
    np = 0x14cbb00}}

Reply:
as i said, it'll get cleared after server_info is deleted via 'delete', i did run valgrind to confirm that there's no memory loss here.

Reply:
we are good then.